app/assets/javascripts/test/backend-snapshot-tests/teamstructure.e2e.js

```diff
@@ -0,0 +1,170 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+/* eslint-disable import/first */
+// @flow
+import {
+  tokenUserDefault,
+  tokenUserAAB,
+  tokenUserAAC,
+  tokenUserABA,
+  tokenUserBAA,
+  setCurrToken,
+} from "../enzyme/e2e-setup";
+import test from "ava";
+import _ from "lodash";
+import * as api from "admin/admin_rest_api";
+
+/*
+TEAM STRUCTURE USED FOR TESTING:
+
+Organizations:
+Connectomics department (Organization A), OrganizationB
+
+Teams:
+Connectomics department (Team AA), Team AB, Team BA, test1, test2
+(the teams test1 and test2 existed before)
+
+User:
+default (UserAAA), User AAB, User AAC, User ABA, User BAA
+
+++----------------------------------------------------------------------------------------------------------------------++--------------------------------------++
+||	Connectomics department (Organization A)																			                                      ||	Organization B 						          ||
+||  Admin: scmboy																										                                                    ||  Admin: userBAA					          	||
+++-----------------------------------+--------------------------+---------------------------+---------------------------++--------------------------------------++
+|| Connectomics department (Team AA) | 		Team AB				        | 		test1				          | 		test2				          ||  TeamBA 								              ||
+++-----------------------------------+--------------------------+---------------------------+---------------------------++--------------------------------------++
+|| scmboy (userAAA)	(teamMng)		     | 		userABA (teamMng.)    | 		scmboy            	  | 		scmboy (teamMng.)	    ||  userBAA	(teamMng.)			         		||
+|| userAAB (teamMng.)		   		       | 						            	| 					              	| 						            	||  							                  		||
+|| userAAC					   		           | 						            	| 					            		| 						            	||  							                  		||
+++-----------------------------------+--------------------------+---------------------------+---------------------------++--------------------------------------++
+ */
+
+// Teams
+test("teams_userDefault", async t => {
+  await setCurrToken(tokenUserDefault);
+  const teams = _.sortBy(await api.getTeams(), team => team.name);
+  t.snapshot(teams[0], { id: "teams_userDefault" });
```

Comment:
The snapshot method is especially useful when testing a large structure for which it is cumbersome to assert meaningful properties (or in the case where you want to ensure that *nothing* changed). As an example, we mostly use this for snapshotting DOM trees. When the backend changes we know that such snapshots should not change.

For the teams api, however, I would not use this method as it can easily break. For example, if the team object gets a new field in the future, this test will break.
Additionally, it's not really clear what exactly is expected in this test. What is the `teams[0]` supposed to be? Team AA? Should the other teams of Organization A also be included in `teams`? Snapshot tests will *always* succeed if you run them for the very first time (as the first run will create the first version of the snapshot). That's why it is important to verify the data in a more explicit way, too (here, I'd avoid snapshots altogether).
An example of how the test could look like with assertions:

```
...
t.is(teams[0].name, "Team AA");
t.is(teams[1].name, "Team AB");
t.is(teams.length, 5);
```

That way, we make a few "spot checks" which also make the intent of the test clear. Such a test would be rather robust against minor changes in the API.